getgauge/registry.py

```diff
@@ -209,6 +211,21 @@ def clear(self):
             setattr(self, '__{}'.format(hook), [])
 
 
+def paths_equal(p1: Union[str, Path], p2: Union[str, Path]) -> bool:
+    """
+    Compare two paths in a cross-platform safe way.
+    On Windows: case-insensitive, slash-insensitive.
+    On Linux/macOS: case-sensitive.
+    """
+    p1 = Path(p1).resolve()
+    p2 = Path(p2).resolve()
+    if sys.platform.startswith("win"):
+        # As Windows is case-insensitive, we can use 'normcase' to compare paths!
+        return os.path.normcase(str(p1)) == os.path.normcase(str(p2))
+    # Mac (and others) allows to use case-sensitive files/folders!
+    return p1 == p2
```

Comment:
It may be the right thing to do, but these operations probably require filesystem operations (since not using `PurePath`, and no way to normalize symlinks/junction points without FS access).

This will likely make things slower. If we can achieve the goal correctly & securely, without resolving actual file system paths we should try to do that.

Also, do we need to mix `os.path` and `pathlib` usage?

1. are you intentionally trying to add normalization of `..` and symlinks via `resolve()`?
2. if not, why not just use normcase for all platforms? [Docs](https://docs.python.org/3/library/os.path.html#os.path.normcase) say it's a no-op on case-sensitive filesystems, so it'd be cleaner to use the library to avoid platform specific logic.